File: backend/Turistando/Turistando.WebApi/Models/ReservaRoteiro.cs
namespace Turistando.WebApi.Models
{
    public class ReservaRoteiro
    {
        public ReservaRoteiro()
        {
            
        }

        public int IdReserva { get; set; }
        public int IdRoteiro { get; set; }
        public string Informacoes { get; set; }
        public double Valor { get; set; }
        public double Avaliacao { get; set; }

        public virtual Reserva Reserva { get; set; }
        public virtual Roteiro Roteiro { get; set; }
    }
}

File: frontend/turistando-react/src/models/reserva-roteiro.ts
import {Reserva} from "./reserva";
import {Roteiro} from "./roteiro";

export class ReservaRoteiro
{
    constructor(idReserva: number, idRoteiro: number)
    {
        this.IdReserva = idReserva;
        this.IdRoteiro = idRoteiro;
        this.Informacoes = '';
        this.Valor = 0;
        this.Avaliacao = 0;
    }

    public IdReserva: number;
    public IdRoteiro: number;
    public Informacoes: string;
    public Valor: number;
    public Avaliacao: number;

    public Reserva: Reserva;
    public Roteiro: Roteiro;
}
